src/analysis/inner_product_space/adjoint.lean

```diff
@@ -186,6 +186,37 @@ lemma is_adjoint_pair (A : E' →L[ℝ] F') :
 
 end real
 
+lemma norm_map_of_unitary {U : (E →L[𝕜] E)} (hU : U ∈ unitary (E →L[𝕜] E)) (x : E) :
+  ∥U x∥ = ∥x∥ :=
```

Comment:
I think the best way to show this would be to construct a linear isometric equivalence (i.e. a `E ≃ₗᵢ[𝕜] E`) from a unitary element of `E →L[𝕜] E`.

Reply:
I think you are suggesting that the lemma be replaced with this construction, correct?

Happy to do so, but I want to clarify.

Reply:
More or less: you could construct the object, and then use it to prove this lemma (at which point it should be a one-liner).

Reply:
I have added `linear_equiv_of_unitary` and `linear_isometry_equiv_of_unitary`, and I think they do what you suggest.  I was not able to set it up in a way so that `norm_map_of_unitary` is quite a one-liner, which makes me think that I'm doing at least one thing the hard way.

Reply:
I think having the `≃ₗᵢ` is a good idea, but I don't see the point of using it to prove this lemma when the proof is identical anyway.